tests/unit/Owncloud/OcisPhpSdk/EducationUserTest.php

```diff
@@ -2,41 +2,182 @@
+        );
+        $ocis = new Ocis(serviceUrl: 'a', accessToken: 'a', educationAccessToken: 'a');
+        /** @phan-suppress-next-line PhanTypeMismatchArgument*/
+        $user = $ocis->getEducationUserById("user-id", apiInstance: $educationApi);
         $this->assertInstanceOf(EducationUser::class, $user);
+        $this->assertSame("id", $user->getId());
+        $this->assertSame("demo", $user->getDisplayName());
+        $this->assertSame("mail@mail.com", $user->getMail());
+        $this->assertSame("surname", $user->getSurname());
+        $this->assertSame("givenname", $user->getGivenName());
+        $this->assertSame("manager", $user->getPrimaryRole());
+        $identities = $user->getidentities();
+        $this->assertNotNull($identities);
+        $this->assertCount(1, $identities);
+        $this->assertContainsOnlyInstancesOf(ObjectIdentity::class, $identities);
+        $this->assertSame("idp", $identities[0]->getIssuer());
+        $this->assertSame("idpId", $identities[0]->getIssuerAssignedId());
+    }
+
+    /**
+     * @dataProvider educationUserData
+     *
+     * @param array<int, array<int, array<string, array<int, string>|string>>> $data
+     * @return void
+     */
+    public function testCreateEducationUsers(array $data)
+    {
+        $educationApi = $this->createMock(EducationUserApi::class);
+        $educationApi->method("createEducationUser")->willReturn(
+            new EU($data),
+        );
+        $ocis = new Ocis(serviceUrl: 'a', accessToken: 'a', educationAccessToken: 'a');
+        /** @phan-suppress-next-line PhanTypeMismatchArgument*/
+        $user = $ocis->createEducationUser(displayName: 'demo', onPremisesSAMAccountName: 'demo', apiInstance: $educationApi);
+        $this->assertInstanceOf(EducationUser::class, $user);
+        $this->assertSame("id", $user->getId());
+        $this->assertSame("demo", $user->getDisplayName());
+        $this->assertSame("mail@mail.com", $user->getMail());
+        $this->assertSame("surname", $user->getSurname());
+        $this->assertSame("givenname", $user->getGivenName());
+        $this->assertSame("manager", $user->getPrimaryRole());
+        $identities = $user->getidentities();
+        $this->assertNotNull($identities);
+        $this->assertCount(1, $identities);
+        $this->assertContainsOnlyInstancesOf(ObjectIdentity::class, $identities);
+        $this->assertSame("idp", $identities[0]->getIssuer());
+        $this->assertSame("idpId", $identities[0]->getIssuerAssignedId());
+    }
+
+    /**
+     * @dataProvider educationUserData
+     *
+     * @param array<int, array<int, array<string, array<int, string>|string>>> $data
+     * @return void
+     */
+    public function testGetEducationUserWithOutToken(array $data)
+    {
+        $educationApi = $this->createMock(EducationUserApi::class);
+        $this->expectExceptionMessage("This function cannot be used because no authentication token was provided for the educationUser endpoints.");
```

Comment:
```suggestion
        $this->expectExceptionMessage(
            "This function cannot be used because no authentication token was provided for the educationUser endpoints.
       ");
```